ui/lib/core/addon/components/confirmation-modal.hbs

```diff
@@ -21,21 +13,21 @@
         class="input has-margin-top"
         autocomplete="off"
         spellcheck="false"
-        data-test-confirmation-modal-input={{this.testSelector}}
+        data-test-confirmation-modal-input={{@testSelector}}
```

Comment:
Since test selectors are stripped out of production builds I'm not sure that dedicating a component argument to a test selector value is best. What about spreading the attributes on either the `Modal` or `section` element? You could then remove the value for `data-test-confirmation-modal-input` selector and target it by first selecting the top level element:

```js
<ConfirmationModal data-test-confirm-modal="some-value" />
// confirmation-modal.hbs
<Modal ...attributes></Modal>
// in a test
assert.dom('[data-test-confirm-modal="some-value"] [data-test-confirmation-modal-input]').exists();
```

Reply:
This is a good call. I think in other components I've just passed in an existing attr to help differentiate, I'll update to do something similar here